cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/sink/RealtimeCubeSink.java

```diff
@@ -64,62 +63,90 @@
     "one will be created.";
-  private static final String PROPERTY_RESOLUTIONS_DESC = "Aggregation resolutions. See Cube dataset configuration " +
+  private static final String PROPERTY_RESOLUTIONS_DESC = "Aggregation resolutions to be used if new Cube dataset " +
+    "needs to be created. See Cube dataset configuration " +
     "details for more information : http://docs.cask.co/cdap/current/en/developers-manual/building-blocks/datasets/" +
     "cube.html#cube-configuration";
-  private static final String MAPPING_CONFIG_PROPERTY_DESC = "The StructuredRecord to CubeFact mapping " +
-    "configuration. More info on the format of this configuration can be found at http://docs.cask.co/cdap/current/" +
-    "en/reference-manual/javadocs/co/cask/cdap/template/etl/common/StructuredRecordToCubeFact.html";
-
-  private static final String CUSTOM_PROPERTIES_DESC = "Provide any custom properties (such as Aggregations) as a " +
-    "JSON Map. For example if aggregations are desired on fields - abc and xyz, the property should have the value : " +
-    "\"{\"dataset.cube.aggregation.agg1.dimensions\":\"abc\", \"dataset.cube.aggregation.agg2.dimensions\":\"xyz\"}";
+  private static final String DATASET_OTHER_PROPERTIES_DESC = "Provide any dataset properties to be used if new Cube " +
+    "dataset needs to be created as a JSON Map. For example if aggregations are desired on fields - abc and xyz, the " +
+    "property should have the value: " +
+    "\"{\"dataset.cube.aggregation.agg1.dimensions\":\"abc\", \"dataset.cube.aggregation.agg2.dimensions\":\"xyz\"}." +
+    " See Cube dataset configuration " +
+    "details for more information : http://docs.cask.co/cdap/current/en/developers-manual/building-blocks/datasets/" +
+    "cube.html#cube-configuration";
+  private static final String FACT_TS_FIELD_PROPERTY_DESC = "Name of the StructuredRecord's field that contains" +
+    " timestamp to be used in CubeFact. If not provided, the current time of the record processing will be used as " +
+    "CubeFact timestamp.";
+  private static final String FACT_TS_FORMAT_PROPERTY_DESC = "Format of the value of timstamp field, " +
+    "e.g. \"HH:mm:ss\" (used if " + Properties.Cube.FACT_TS_FIELD + " is provided).";
+  private static final String MEASUREMENTS_PROPERTY_DESC = "Measurements to be extracted from StructuredRecord to be" +
+    " used in CubeFact. Provide properties as a JSON Map. For example use 'price' field as a measurement of type" +
+    " gauge, and 'count' field as a measurement of type counter, the property should have the value: " +
+    "\"{\"cubeFact.measurement.price\":\"GAUGE\", \"cubeFact.measurement.count\":\"COUNTER\"}.";
 
   /**
-   * Config class for RealtimeCube
+   * Config class for RealtimeCubeSink
    */
-  public static class RealtimeCubeConfig extends PluginConfig {
+  public static class RealtimeCubeSinkConfig extends PluginConfig {
```

Comment:
Is the RealtimeCubeSinkConfig any different from BatchCubeSinkConfig? It looks like we can share most of the config, description etc. DBConfig is an example of how a common config is shared between two plugins.